### PR TITLE
fix(freebsd.plugin): fix wired/cached/avail memory calculation on FreeBSD with ZFS

### PR DESCRIPTION
**Pull request by ilyam8**
##### Summary

Fixes: #13172

##### Test Plan

Install on FreeBSD with ZFS, check the `system.ram` and `mem.available` charts.

- wired

Screenshot 2022-06-20 at 20 38 09

- avail (51TB => 13GB)

Screenshot 2022-06-20 at 20 57 31

##### Additional Information

<details> <summary>For users: How does this change affect me?</summary>
  
</details>

### CODE REVIEW

File: collectors/freebsd.plugin/freebsd_sysctl.c
Comment (ilyam8) [inline collectors/freebsd.plugin/freebsd_sysctl.c:1082]: @MrZammler @vlvkobal @thiagoftsm check if we need to case to a different type here, I don't know.

- vmmeter_data.* is `unsigned int`
- zfs_arcstats_shrinkable_cache_size_bytes is `long long int`
- system_pagesize is `int`

File: collectors/freebsd.plugin/freebsd_sysctl.c
Comment (Dimitris-P) [inline collectors/freebsd.plugin/freebsd_sysctl.c:1050]: Shouldn't the cast be to `collected_number` type?

Comment (ilyam8) [reply collectors/freebsd.plugin/freebsd_sysctl.c:1050]: I think there is the implicit casting: calculation result => collected_number when passed to the func. Not a specialist (not c dev).

File: collectors/freebsd.plugin/freebsd_sysctl.c
Comment (thiagoftsm) [reply collectors/freebsd.plugin/freebsd_sysctl.c:1082]: Hello @ilyam8 ,

In theory we are working with `integers` here, but you address important points:

- The biggest difference for `unsigned int` and `int` it is the way one bit is interpreted, it is a number for the first and a signal for the second.
- Now when we talk about `long long int` and `int` they have size difference between them.

I would suggest to convert everything to `long long int` to be sure that nothing would be lost. I would not expect that all compilers will require you to cast the data, `gcc 12`, for example, does not say nothing.

Best regards!

File: collectors/freebsd.plugin/freebsd_sysctl.c
Comment (Dim-P) [reply collectors/freebsd.plugin/freebsd_sysctl.c:1050]: Yep, that's what I had in mind. You may get a compiler warning in some cases.

**Comment by thiagoftsm** [comment]
Hello @ilyam8 ,

On my environment everything looks fine, I am waiting for your decision about casts after the explanation to approve your PR.

**Review by vlvkobal — APPROVED** [review]

File: collectors/freebsd.plugin/freebsd_sysctl.c
Comment (ilyam8) [reply collectors/freebsd.plugin/freebsd_sysctl.c:1050]: > some compilers can create warnings, mainly the oldest one

is not a strong reason, I am not "fixing" it in this PR. If you check other "rrddim_set_by_pointer" calls in this collector you see there are no casting to "collected_number". If you believe this is needed consider creating a separate PR.

**Pull request merged**
